powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/EciesEnvelopeKey.java

```diff
@@ -85,6 +86,9 @@ public EciesEnvelopeKey(byte[] secretKey, byte[] ephemeralPublicKey) {
      */
     static EciesEnvelopeKey fromPublicKey(PublicKey publicKey, byte[] sharedInfo1) throws EciesException {
         try {
+            if (keyGenerator == null) {
```

Comment:
... this looks a bit weird.

Reply:
Yes, I know, it is caused by static context. Anyway I think we can handle all these style issues by initializing the SecureRandom lazily.